src/tools/run-make-support/src/lib.rs

```diff
@@ -169,6 +169,29 @@ pub fn cwd() -> PathBuf {
     env::current_dir().unwrap()
 }
 
+// FIXME(Oneirical): This will no longer be required after compiletest receives the ability
+// to manipulate read-only files. See https://github.com/rust-lang/rust/issues/126334
+/// Ensure that the path P is read-only while the test runs, and restore original permissions
+/// at the end so compiletest can clean up.
+#[track_caller]
+pub fn test_while_readonly<P: AsRef<Path>, F: FnOnce() + std::panic::UnwindSafe>(
```

Comment:
Question: won't this be very misleading on Windows? Maybe this needs to check if non-Windows and link the doc comment for `Permissions::set_readonly(true)` and/or explain that `set_readonly` is ineffective for Windows directory itself or preventing directory content modifications?

Reply:
Added a check that verifies if the user is simultaneously on Windows and using test_while_readonly on a dir, causing a detailed error message and a panic.
@rustbot review

Reply:
Thanks, could we please document that it will panic on Windows if path is a directory?